sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala

```diff
@@ -0,0 +1,210 @@
+        "CREATE TABLE \"test\".\"empty_table\" (name TEXT(32) NOT NULL, id INTEGER NOT NULL)")
+        .executeUpdate()
+      conn.prepareStatement(
+        "CREATE TABLE \"test\".\"people\" (name TEXT(32) NOT NULL, id INTEGER NOT NULL)")
+        .executeUpdate()
+      conn.prepareStatement("INSERT INTO \"test\".\"people\" VALUES ('fred', 1)").executeUpdate()
+      conn.prepareStatement("INSERT INTO \"test\".\"people\" VALUES ('mary', 2)").executeUpdate()
+    }
+  }
+
+  override def afterAll(): Unit = {
+    Utils.deleteRecursively(tempDir)
+    super.afterAll()
+  }
+
+  test("simple scan") {
+    checkAnswer(sql("SELECT * FROM h2.test.empty_table"), Seq())
+    checkAnswer(sql("SELECT * FROM h2.test.people"), Seq(Row("fred", 1), Row("mary", 2)))
+    checkAnswer(sql("SELECT name, id FROM h2.test.people"), Seq(Row("fred", 1), Row("mary", 2)))
+  }
+
+  test("scan with filter push-down") {
+    val df = spark.table("h2.test.people").filter($"id" > 1)
+    val filters = df.queryExecution.optimizedPlan.collect {
+      case f: Filter => f
+    }
+    assert(filters.isEmpty)
+    checkAnswer(df, Row("mary", 2))
+  }
+
+  test("scan with column pruning") {
+    val df = spark.table("h2.test.people").select("id")
+    val scan = df.queryExecution.optimizedPlan.collectFirst {
+      case s: DataSourceV2ScanRelation => s
+    }.get
+    assert(scan.schema.names.sameElements(Seq("ID")))
+    checkAnswer(df, Seq(Row(1), Row(2)))
+  }
+
+  test("scan with filter push-down and column pruning") {
+    val df = spark.table("h2.test.people").filter($"id" > 1).select("name")
+    val filters = df.queryExecution.optimizedPlan.collect {
+      case f: Filter => f
+    }
+    assert(filters.isEmpty)
+    val scan = df.queryExecution.optimizedPlan.collectFirst {
+      case s: DataSourceV2ScanRelation => s
+    }.get
+    assert(scan.schema.names.sameElements(Seq("NAME")))
+    checkAnswer(df, Row("mary"))
+  }
+
+  // TODO (SPARK-32592): this doesn't work because `DataFrameReader.table` ignores the options.
+  ignore("scan with partition info") {
+    val df = spark.read
+      .option("partitionColumn", "id")
+      .option("lowerBound", "0")
+      .option("upperBound", "3")
+      .option("numPartitions", "2")
+      .table("h2.test.people")
+    assert(df.rdd.getNumPartitions == 2)
+  }
+
+  test("show tables") {
+    checkAnswer(sql("SHOW TABLES IN h2.test"),
+      Seq(Row("test", "people"), Row("test", "empty_table")))
+  }
+
+  // TODO (SPARK-32603): Operation not allowed: CREATE TABLE ... STORED AS ... does not support
+  // multi-part identifiers
+  ignore("create/drop table") {
+    sql("CREATE TABLE h2.test.abc(i INT, j STRING)")
+    checkAnswer(sql("SHOW TABLES IN h2.test"), Seq(Row("test", "people"), Row("test", "abc")))
+    sql("DROP TABLE h2.test.abc")
+    checkAnswer(sql("SHOW TABLES IN h2.test"), Row("test", "people"))
+  }
+
+  // TODO (SPARK-32603): Operation not allowed: CREATE TABLE ... STORED AS ... does not support
```

Comment:
let's follow `JDBCTableCatalogSuite` and use `USING _` to bypass this limitation.